File: projects/.NETCore/Benchmarks/SerializationJson.cs
using BenchmarkDotNet.Attributes;

namespace Benchmarks
{
    public class SerializationJson
    {
        private proto.Account _account;
        private string _json;

        public SerializationJson()
        {
            // Create a new account with some orders
            _account = proto.Account.Default;
            _account.id = 1;
            _account.name = "Test";
            _account.state = proto.State.good;
            _account.wallet.currency = "USD";
            _account.wallet.amount = 1000.0;
            _account.asset = new proto.Balance("EUR", 100.0);
            _account.orders.Add(new proto.Order(1, "EURUSD", proto.OrderSide.buy, proto.OrderType.market, 1.23456, 1000.0));
            _account.orders.Add(new proto.Order(2, "EURUSD", proto.OrderSide.sell, proto.OrderType.limit, 1.0, 100.0));
            _account.orders.Add(new proto.Order(3, "EURUSD", proto.OrderSide.buy, proto.OrderType.stop, 1.5, 10.0));

            // Serialize the account to the JSON string
            _json = _account.ToJson();

            // Deserialize the account from the JSON string
            _account = proto.Account.FromJson(_json);
        }

        [Benchmark]
        public void Serialize()
        {
            // Serialize the account to the JSON string
            _json = _account.ToJson();
        }

        [Benchmark]
        public void Deserialize()
        {
            // Deserialize the account from the JSON string
            _account = proto.Account.FromJson(_json);
        }
    }
}

File: projects/Swift/Tests/Examples/SerializationJson.swift
//
//  SerializationJson.swift
//

import XCTest
import ChronoxorProto

class ExampleSerializationJson: XCTestCase {
    func testSerializationJson() {
        // Create a new account with some orders
        var account = Account(id: 1, name: "Test", state: State.good, wallet: Balance(currency: "USD", amount: 1000.0), asset: Balance(currency: "EUR", amount: 100.0), orders: [])
        account.orders.append(Order(id: 1, symbol: "EURUSD", side: OrderSide.buy, type: OrderType.market, price: 1.23456, volume: 1000.0))
        account.orders.append(Order(id: 2, symbol: "EURUSD", side: OrderSide.sell, type: OrderType.limit, price: 1.0, volume: 100.0))
        account.orders.append(Order(id: 3, symbol: "EURUSD", side: OrderSide.buy, type: OrderType.stop, price: 1.5, volume: 10.0))

        // Serialize the account to the JSON string
        let json = try! account.toJson()

        // Show the serialized JSON and its size
        print("JSON: \(json)")
        print("JSON size: \(json.count)")

        // Deserialize the account from the JSON string
        account = try! Account.fromJson(json)

        // Show account content
        print("")
        print(account.description)
    }
}
